addons/binding/org.openhab.binding.solaredge/src/main/java/org/openhab/binding/solaredge/handler/SolarEdgeBaseHandler.java

```diff
@@ -0,0 +1,181 @@
+
+    /**
+     * Schedule for polling
+     */
+    private ScheduledFuture<?> pollingJob;
+
+    public SolarEdgeBaseHandler(@NonNull Thing thing) {
+        super(thing);
+        this.pollingRunnable = new SolarEdgePolling(this);
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        logger.debug("command for {}: {}", channelUID, command);
+
+        // TODO: write access is not supported.
+    }
+
+    @Override
+    public void initialize() {
+        logger.debug("About to initialize SolarEdge");
+        Thing thing = this.getThing();
+        SolarEdgeConfiguration config = getConfiguration();
+
+        logger.debug("Discovered NibeUplink initialized: {}", config);
+
+        this.refreshInterval = config.getPollingInterval();
+        this.webInterface = new WebInterface(config, this);
+
+        if (config.getPassword() != null && config.getUsername() != null) {
+            this.startPolling();
+        } else {
+            thing.setStatusInfo(new ThingStatusInfo(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
+                    "no username/password set"));
+        }
+    }
+
+    /**
+     * Start the polling.
+     */
+    private synchronized void startPolling() {
+        if (pollingJob == null || pollingJob.isCancelled()) {
+            logger.debug("start polling job at intervall {}", refreshInterval);
+            pollingJob = scheduler.scheduleWithFixedDelay(pollingRunnable, 1, refreshInterval, TimeUnit.SECONDS);
+        } else {
+            logger.debug("pollingJob already active");
+        }
+    }
+
+    /**
+     * Disposes the bridge.
+     */
+    @Override
+    public void dispose() {
+        logger.debug("Handler disposed.");
+        if (pollingJob != null && !pollingJob.isCancelled()) {
+            logger.debug("stop polling job");
+            pollingJob.cancel(true);
+            pollingJob = null;
+        }
+    }
+
+    @Override
+    public WebInterface getWebInterface() {
+        return webInterface;
+    }
+
+    /**
+     * will update all channels provided in the map
+     */
+    @Override
+    public void updateChannelStatus(Map<String, String> values) {
+        logger.debug("Handling channel update.");
+
+        for (String key : values.keySet()) {
+            Channel channel = getThingSpecificChannel(key);
+            if (channel != null) {
+                String value = values.get(key);
+                logger.debug("Channel is to be updated: {}: {}", channel.getFQName(), value);
+                if (value != null && !value.equals(NO_VALUE)) {
+                    if (channel.getJavaType().equals(Double.class)) {
+                        try {
+                            updateState(channel.getFQName(), convertToDecimal(value));
+                        } catch (NumberFormatException ex) {
+                            logger.warn("Could not update channel {} - invalid number: '{}'", channel.getFQName(),
```

Comment:
Maybe also set the channel state to UNDEF? Same with the else case below when the value is null?

Reply:
is there any constant for "UNDEF"? setting 'new StringType("UNDEF")' should not work in case the type is not String.

Reply:
`UnDefType.UNDEF` should work for all types.

Reply:
I'd say it depends on what we have here: Is it a software bug that must not happen? If so, add the exception to the log message and do NOT do any state updates (as we encountered a bug anyhow). If it is a valid response from the server, which we cannot express as a state here, then only do debug logging and set the state to UNDEF.